central/graphql/resolvers/image_vulnerabilities.go

```diff
@@ -707,3 +805,335 @@ func (resolver *imageCVEResolver) SuppressExpiry(_ context.Context) (*graphql.Ti
+	return nil, nil
+}
+
+func (resolver *imageCVEV2Resolver) SuppressExpiry(_ context.Context) (*graphql.Time, error) {
+	return nil, nil
+}
+
+func (resolver *imageCVEV2Resolver) Suppressed(_ context.Context) bool {
+	return false
+}
+
+func (resolver *imageCVEV2Resolver) EnvImpact(ctx context.Context) (float64, error) {
+	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.ImageCVEs, "EnvImpact")
+	allCount, err := resolver.root.DeploymentCount(ctx, RawQuery{})
+	if err != nil || allCount == 0 {
+		return 0, err
+	}
+	ctx = scoped.Context(ctx, scoped.Scope{
+		ID:    resolver.data.GetId(),
+		Level: v1.SearchCategory_IMAGE_VULNERABILITIES_V2,
+	})
+	scopedCount, err := resolver.root.DeploymentCount(ctx, RawQuery{})
+	if err != nil {
+		return 0, err
+	}
+	return float64(scopedCount) / float64(allCount), nil
+}
+
+func (resolver *imageCVEV2Resolver) FixedByVersion(ctx context.Context) (string, error) {
+	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.ImageCVEs, "FixedByVersion")
+	if resolver.ctx == nil {
+		resolver.ctx = ctx
+	}
+
+	// Short path. Full image is embedded when image scan resolver is called.
+	if embeddedVuln := embeddedobjs.VulnFromContext(resolver.ctx); embeddedVuln != nil {
+		return embeddedVuln.GetFixedBy(), nil
+	}
+
+	scope, hasScope := scoped.GetScope(resolver.ctx)
+	if !hasScope {
+		return "", nil
+	}
+	if scope.Level != v1.SearchCategory_IMAGE_COMPONENTS_V2 {
+		return "", nil
+	}
+
+	query := search.NewQueryBuilder().AddExactMatches(search.CVEID, resolver.data.GetId()).ProtoQuery()
+	cves, err := resolver.root.ImageCVEV2DataStore.SearchRawImageCVEs(resolver.ctx, query)
+	if err != nil || len(cves) == 0 {
+		return "", err
+	}
+	return cves[0].GetFixedBy(), nil
+}
+
+// IsFixable returns if the CVE is fixable or not.
+//
+//	TODO(ROX-28123): Once the old code is removed, this method can become generated.
+func (resolver *imageCVEV2Resolver) IsFixable(_ context.Context, _ RawQuery) (bool, error) {
+	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.ImageCVEs, "IsFixable")
+	return resolver.data.IsFixable, nil
+}
+
+func (resolver *imageCVEV2Resolver) LastScanned(ctx context.Context) (*graphql.Time, error) {
+	defer metrics.SetGraphQLOperationDurationTime(time.Now(), pkgMetrics.ImageCVEs, "LastScanned")
+
+	// Short path. Full image is embedded when image scan resolver is called.
+	if scanTime := embeddedobjs.LastScannedFromContext(resolver.ctx); scanTime != nil {
+		return &graphql.Time{Time: *scanTime}, nil
+	}
+
+	imageLoader, err := loaders.GetImageLoader(resolver.ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	q := search.EmptyQuery()
+	q.Pagination = &v1.QueryPagination{
+		Limit:  1,
+		Offset: 0,
+		SortOptions: []*v1.QuerySortOption{
+			{
+				Field:    search.ImageScanTime.String(),
+				Reversed: true,
+			},
+		},
+	}
+
+	images, err := imageLoader.FromQuery(resolver.imageVulnerabilityScopeContext(ctx), q)
```

Comment:
Same comment as in `LastScanned` resolver in image_components. Can get image ID from resolver.data and directly query for that image ID.